app/person-escort-record/controllers/framework-step.js

```diff
@@ -93,6 +93,10 @@ class FrameworkStepController extends FormWizardController {
       // call parent saveValues to handle storing new values in the session
       super.saveValues(req, res, next)
     } catch (error) {
+      if (req.body['js-items-length']) {
```

Comment:
What is the error instance when this occurs? It feels a little odd to use a condition that checks for something in the request body to determine whether to ignore the error or not.

Would it not be better to maybe at least extend the condition for the error instance in which it occurs?

Reply:
Also, redirecting straight away will skip saving the values to the session, is that the desired effect?

Reply:
If a user ends up with fewer fields than expected or more fields than expected but all the info filled in with no validation errors, everything just works.

This covers the case where a user has added more fields than expected and then does not correctly fill out those fields (or invalidates an existing field). The PER record is rejected with a 422 unprocessable error, but if the user is simply redirected back to the original page as a get, the PER has now been updated to have the correct number of fields... and so can display the expected error messages (which it cannot if the user is simply presented with a 422 error) - the values do get saved to the session and do persist

- Yes, it should indeed be added in to not do the same for unexpected errors (when the js adds are in effect, that is)
- Yes, it would be nice/better to fix the PER BE to actually handle this, but...
- Yes, it's odd.

I did also try to actually add and remove the fields as ajax-style calls when the user clicks the buttons but that gets way more complicated

1. CSRF tokens
2. Identifying all the correct form fields and their values that need to be sent
3. Handling the cases where the items fail to get added or removed by the backend
4. Handling the cases where we've failed to keep the client and server version in sync

But I may have missed a trick there

Reply:
This must actually have been some side-effect of bumpiness in the backend. I don't see the 422s any more and there's no need to do anything special